src/WebJobs.Script.Host/WebJobs.Script.Host.csproj

```diff
@@ -176,7 +197,9 @@
       <ErrorText>This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
     </PropertyGroup>
     <Error Condition="!Exists('..\..\packages\StyleCop.MSBuild.4.7.50.0\build\StyleCop.MSBuild.Targets')" Text="$([System.String]::Format('$(ErrorText)', '..\..\packages\StyleCop.MSBuild.4.7.50.0\build\StyleCop.MSBuild.Targets'))" />
+    <Error Condition="!Exists('..\..\packages\Microsoft.Bcl.Build.1.0.21\build\Microsoft.Bcl.Build.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\..\packages\Microsoft.Bcl.Build.1.0.21\build\Microsoft.Bcl.Build.targets'))" />
   </Target>
+  <Import Project="..\..\packages\Microsoft.Bcl.Build.1.0.21\build\Microsoft.Bcl.Build.targets" Condition="Exists('..\..\packages\Microsoft.Bcl.Build.1.0.21\build\Microsoft.Bcl.Build.targets')" />
```

Comment:
Yes it has to be there. Mobile Services client requires it. This piece is added via nuget -- I do nothing magic here. 

Mobile Services needs it b/c of its dependency on Microsoft.Bcl -- I believe it's related to the Portable Class Libraries (since it targets a bunch of platforms).

The nuproj is the weird bit. This Bcl.Build package spits warnings if downstream projects do not reference it so you need to reference it there so the project dependencies are discovered properly.

I now know how to test it locally -- see your mail. Pass /m when building locally to truly replicate with TeamCity is doing.

---
In reply to: [56189477](https://github.com/Azure/azure-webjobs-sdk-script/pull/98#discussion_r56189477) [](ancestors = 56189477)